.github/workflows/e2e.yml

```diff
@@ -54,14 +57,123 @@ jobs:
         run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- pnpm wallet playwright test
       - uses: actions/upload-artifact@v3
         if: always()
         with:
           name: playwright-report-wallet
           path: apps/wallet/playwright-report/
           retention-days: 30
+
+  # Run e2e test against localnet built on the devnet branch for backward compatibility check
+  local_devnet_branch:
+    name: Local Network Built on devnet branch
+    needs: diff
+    # TODO: add wallet e2e to the `if` condition when available
+    if: needs.diff.outputs.isExplorer == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'
+    runs-on: ubuntu-ghcloud
+    steps:
+      # checkout the devnet branch
+      - uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
+        with:
+          ref: devnet
+      - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # pin@v1
+      - uses: bmwill/rust-cache@v1 # Fork of 'Swatinem/rust-cache' which allows caching additional paths
+      - uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # pin@v2.2.4
+        with:
+          version: 7
+      - run: cargo build --bin sui-test-validator --bin sui --profile dev
+
+      # checkout the current branch
+      - uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
+        with:
+          clean: false
+
+      - name: Install Nodejs
+        uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # pin@v3
+        with:
+          node-version: "18"
+          cache: "pnpm"
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+      - name: Install Playwright Browsers
+        run: pnpm explorer playwright install --with-deps chromium
+
+      - name: Set env
+        run: echo "E2E_RUN_LOCAL_NET_CMD=(RUST_LOG=\"consensus=off\" $(echo $PWD/target/debug/sui-test-validator))" >> $GITHUB_ENV
+
+      - name: Run TS SDK e2e tests
+        if: ${{ needs.diff.outputs.isTypescriptSDK == 'true' }}
+        run: pnpm dlx concurrently --kill-others --success command-1 "$E2E_RUN_LOCAL_NET_CMD" 'pnpm sdk test:e2e'
+
+      - name: Run Explorer e2e tests
+        if: ${{ needs.diff.outputs.isExplorer == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'}}
+        run: pnpm explorer playwright test
+
+      # TODO: add wallet e2e when available
+
+      - uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: playwright-report-explorer
+          path: apps/explorer/playwright-report/
+          retention-days: 30
+
+  # Run e2e test against localnet built on the Testnet branch for backward compatibility check
+  local_testnet_branch:
+    name: Local Network Built on testnet branch
+    needs: diff
+    # TODO: add wallet e2e to the `if` condition when available
+    if: needs.diff.outputs.isExplorer == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'
+    runs-on: ubuntu-ghcloud
```

Comment:
Last I checked these tests take 5-10 minutes and we're adding 2 additional runs per PR, so we probably want to validate that we have the CI runner capacity for this. This will get a lot better once we have binaries we can just run too

Reply:
I'll follow up with @ebmifa on this